app/components/Snackbar/index.js

```diff
@@ -30,14 +30,17 @@ const snackbarClasses = ({ type }) => ({
 
 @autobind
 class Snackbar extends React.Component {
-
   constructor(props) {
     super(props);
     this.hideTimer = null;
+    this.newMessage = null;
+    this.timeoutRunning = null;
     this.state = {
       message: props.messages.length > 0
         ? props.messages[props.messages.length-1]
-        : null
+        : null,
+      delay: 250, // delay in ms to render component
+      isWaiting: true,
```

Comment:
You can drop `isWaiting` (or try again to modify the logic).

Notice that isWaiting is only set to true in one place: the constructor. Therefore, after the first message is received it will never be true again,